### ISSUE TITLE
DeprecatedWarnings when loaded

### ISSUE THREAD

**Issue by sneiman** [PyTorchLightning/pytorch-lightning#1178]
Ubuntu 19.10, Python 3.7.5, pytorch 1.4.0, venv, installed with pip3

Builds based on clones of master after ~3/14 may issue DeprecationWarnings when loaded. This occurs if built within venv environment with `include-system-site-packages = true` or built outside venv. It does not occur if `include-system-site-packages = false`, or if using earlier repo clone. 

This is a fresh ubuntu install, and the only python packages installed are in support of ml work - pytorch, pytorch-lightning, numpy, sci-py, tensorflow, etc.

The errors:
```
/home/seth/dev/pytorch_work/.venv.ptl.master/lib/python3.7/site-packages/graphql/type/directives.py:55: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
  assert isinstance(locations, collections.Iterable), 'Must provide locations for directive.'
/home/seth/dev/pytorch_work/.venv.ptl.master/lib/python3.7/site-packages/graphql/type/typemap.py:1: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
  from collections import OrderedDict, Sequence, defaultdict
/home/seth/dev/pytorch_work/.venv.ptl.master/lib/python3.7/site-packages/trains/backend_interface/metrics/events.py:27: DeprecationWarning: The usage of `cmp` is deprecated and will be removed on or after 2021-06-01.  Please use `eq` and `order` instead.
  @attr.attrs(cmp=False, slots=True)
```
Seems like this might be due to new code that depend on newer versions of installed packages, so here is pip3 freeze outside of venv:
```
numpy==1.18.1
Pillow==6.1.0
torch==1.4.0
torchvision==0.5.0
tqdm==4.43.0
```

**Comment by Borda** [comment]
could you please give full error trace?
so far I see only `trains` error which is third-party lib...
maybe @bmartinn could help...

**Comment by sneiman** [comment]
That is the full  trace. Three warnings come up - 2 for graphql, 1 for trace. Both 3rd party libs. The reason I raised the question is that they did not start appearing until repo clones after ~march 14. I haven't changed the environment ...

**Comment by Borda** [comment]
that is about the time we have added Trains...

**Comment by bmartinn** [comment]
@Borda , I can probably take care of the warning inside the Trains package, but as you mentioned those are 3rd party libraries...

**Comment by sneiman** [comment]
If its not due to new ptl code making an error or using the wrong version of something, it seems like this is at most a release note - easy for users to mask with a filter. Shall we close?

**Comment by Borda** [comment]
@bmartinn kind of extreme case would be using some warning ignore?

**Comment by bmartinn** [comment]
@Borda Doable, but this is a Trains/GraphQL repository issue, don't you think?

**Comment by sneiman** [comment]
fwiw - i think ptl should let the warnings stand if they are not due to some 'mis-behavior' in ptl.  they are easily screened by a warning filter in user code and if trains doesn't fix them before they become fatal at least user will know.

Sent from mobile device - sorry for typos and brevity.

This entire message is confidential. If it isn't intended for you, you may not use it - so please throw it away and forget about it.

On Mar 18, 2020, at 4:09 PM, Martin.B <notifications@github.com> wrote:

﻿

@Borda<https://github.com/Borda> Doable, but this is a Trains/GraphQL repository issue, don't you think?

**Comment by Borda** [comment]
as it is Trains issue, I ll close it here... feel free reopen if needed :rabbit:
